pkg/network/network_test.go

```diff
@@ -490,6 +490,94 @@ func TestConfiguration(t *testing.T) {
 	}
 }
 
+// TODO move this to TestConfiguration
+func TestSubDomain(t *testing.T) {
+	subdomainTemplate := "{{if eq (len .SubDomain) 0}}{{.Name}}.{{.Namespace}}.{{.Domain}}{{ else }}{{.Name}}.{{.SubDomain}}.{{.Domain}}{{ end }}"
+	networkConfigTests := []struct {
+		name       string
+		wantErr    bool
+		wantConfig *Config
+		config     *corev1.ConfigMap
+		data       DomainTemplateValues
+	}{{
+		name:    "network configuration with subdomain in template",
+		wantErr: false,
+		wantConfig: &Config{
+			IstioOutboundIPRanges:      "*",
+			DefaultClusterIngressClass: "foo-ingress",
+			DomainTemplate:             subdomainTemplate,
+			HTTPProtocol:               HTTPEnabled,
+		},
+		config: &corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: system.Namespace(),
+				Name:      ConfigName,
+			},
+			Data: map[string]string{
+				IstioOutboundIPRangesKey:      "*",
+				DefaultClusterIngressClassKey: "foo-ingress",
+				DomainTemplateKey:             subdomainTemplate,
+			},
+		},
+		data: DomainTemplateValues{
+			Name:      "foo",
+			Namespace: "bar",
+			SubDomain: "sub1",
```

Comment:
Update test?

Reply:
Yes :) thought of doing it once everyone agrees